ubi_manifest/app/api.py

```diff
@@ -1,8 +1,150 @@
 async def status():
     return {"status": "OK"}
+
+
+@router.post(
+    "/manifest",
+    response_model=List[TaskState],
+    status_code=201,
+    responses={
+        201: {
+            "description": "Depsolve tasks created",
+            "content": {
+                "application/json": {
+                    "example": [
+                        {
+                            "task_id": "some_task_id",
+                            "state": "PENDING",
+                        }
+                    ]
+                }
+            },
+        },
+        404: {
+            "description": "None of repositories requested are not allowed for depsolving.",
+            "content": {
+                "application/json": {
+                    "example": {
+                        "details": "None of [repo_id] are allowed for depsolving."
+                    }
+                }
+            },
+        },
+    },
+)
+async def manifest_post(depsolve_item: DepsolveItem) -> List[TaskState]:
+    repo_groups = {}
+    # compare provided repo_ids with the config and pick allowed repo groups
+    for repo_id in depsolve_item.repo_ids:
+        for key, group in app.conf.allowed_ubi_repo_groups.items():
+            if repo_id in group:
+                repo_groups.setdefault(key, group)
+
+    if not repo_groups:
+        raise HTTPException(
+            status_code=404,
+            detail=f"None of {depsolve_item.repo_ids} are allowed for depsolving.",
+        )
+
+    tasks_states = []
+    for repo_group in repo_groups.values():
+        task = depsolve_task.apply_async(args=[repo_group])
+        tasks_states.append(TaskState(task_id=task.task_id, state=task.state))
+
+    return tasks_states
+
+
+@router.get(
+    "/manifest/{repo_id}",
+    response_model=DepsolverResult,
+    status_code=200,
+    responses={
+        200: {
+            "description": "Depsolved content for repo_id found",
+            "content": {
+                "application/json": {
+                    "example": {
+                        "repo_id": "foo-bar-repo",
+                        "content": [
+                            {
+                                "src_repo_id": "source-foo-bar-repo",
+                                "unit_type": "RpmUnit",
+                                "unit_attr": "filename",
+                                "value": "some-filename.rpm",
+                            }
+                        ],
+                    }
+                }
+            },
+        },
+        404: {
+            "description": "Content for request repository is not available.",
+            "content": {
+                "application/json": {
+                    "example": {"detail": "Content for foo-repo not found"}
+                }
+            },
+        },
+    },
+)
+async def manifest_get(repo_id: str) -> DepsolverResult:
+    redis_client = redis.from_url(app.conf.backend)
+    value = redis_client.get(repo_id) or ""
```

Comment:
Redis docs says that:
>  Values can be strings (including binary data) of every kind, for instance you can store a jpeg image inside a value. A value can't be bigger than 512 MB.
I guess we shouldn't get anywhere near to that value even with large strings. We might consider using redis set instead of string, where each item holds info of one unit but I think this may lead to many json.dumps|loads calls.
So I'll definitely do some tests on realistic data.

Ref: https://redis.io/topics/data-types-intro#redis-strings